### DISCUSSION TITLE
External Library and Mobile photos duplicate prevention?

### DISCUSSION THREAD

**Discussion by matticas** [immich-app/immich#6714] [Q&A] [answered]
Hi there;

I'm trying to understand the Immich Android app and it's photo uploading process.

I have just finished 2 weeks of external library imports for 2 decades of photos, including creating albums, and now I'm ready to sync my phone permanently.

**Question**; my phone is 2yrs old, and all photos (except 2024) are already loaded into my existing/external library, which is imported into Immich.  If I sync my phone, will Immich check before or after the mobile upload for duplicate?  What i'm trying to avoid is duplicate copies on my external library folder and the immich library folder.  
Is it possible to restrict the upload period of a folder to a date range; in this case, 01-Jan-2024 onwards only?  I have the same problem with wife's phone; a year of photos already in the existing/external library.

Thanks for your time, love this Immich thing, it's brilliant.!

**Comment by bo0tzz** [comment]
External libraries are not considered in the deduplication logic.

**Reply by matticas** [reply]
Ok thanks for that.

Is there anyway to limit the android app to only upload/sync from a certain time frame onwards?

**Comment by matticas** [comment, marked as answer]
Hi, closing this out for those that might stumble upon it.

I'm a fool, I didn't think of the obvious answer; the photos on my phone that were already backup manually to my computer (thus External Library in Immich) were the issue, been duplicated.

Simply; 

1. I moved all files in my phone camera folder (DCIM) that were not 2024, into another folder.
2. Thus my phone camera folder (DCIM) only contains 2024 photos, and is backed up to immich.
3. My existing photos pre-2024 are in my External Library.
